### PR TITLE
Documentation on flashing with other programmers

### PR DESCRIPTION
**Pull request by bbadaboom**
As I couldn't get my hands on a  J-Link EDU programmer (out of stock wherever I checked) I bought a cheap ST-LINK clone.
Took me a while to figure out how to flash to the nRF52840 but finally made it. Thought the community can maybe profit from the documentation on how to flash  the firmware with one of these. ;)

### CODE REVIEW

File: docs/firmware-flashing.md [hunk 1]
```diff
@@ -30,4 +30,42 @@ The nRF52840 can be used as a J-Link programmer if you connect it like this:
...
 Once the programmer is connected and you have downloaded the .hex file from the [release](https://github.com/stanvn/zigbee-plant-sensor/releases) page, you can flash the file onto the microcontroller using [nRF Connect Programmer](https://infocenter.nordicsemi.com/index.jsp?topic=/struct_nrftools/struct/nrftools_nrfconnect.html). There documentation describes in details how to flash the hex file.
+
+## Programming using ST-LINK
+Flashing to the nRF52840 can also be achieved with other programmers like ST-LINK but this is not possible using Nordicsemi's software.
+
+### Prepare for flashing:
+- Download the .hex file from the [release](https://github.com/stanvn/zigbee-plant-sensor/releases) page
+- Make sure your [ST-LINK driver](https://os.mbed.com/teams/ST/wiki/ST-Link-Driver) is correctly installed
+- Make sure you have a tool with which you can open telnet connections (e.g. [PuTTY](https://www.chiark.greenend.org.uk/~sgtatham/putty/putty)/[KiTTY](https://9bis.net/kitty/))
+- Download and install  [OpenOCD](https://gnutoolchains.com/arm-eabi/openocd/)
```
Comment (stanvn) [inline docs/firmware-flashing.md]: https://openocd.org/pages/getting-openocd.html
This link might be useful for installing OpenOCD on linux and MacOS

File: docs/firmware-flashing.md [hunk 2]
```diff
@@ -30,4 +30,42 @@ The nRF52840 can be used as a J-Link programmer if you connect it like this:
...
+- In the "bin" folder of OpenOCD and create a file called "openocd_nrf52.cfg" with the following content:
```
Comment (stanvn) [inline docs/firmware-flashing.md]: I think 'in' should be 'open'? Maybe give the reader some clue where to find this bin folder?

**Comment by stanvn** [comment]
Thanks for updating the docs! This might indeed be very useful for others with a ST-Link. I have only two comments (see above). Once resolved I'll merge the docs. 
Additional tip: it is also possible to use VSCode to flash it with the ST-Link and OpenOCD. I've done it long time ago, but I don't remember the exact details

File: docs/firmware-flashing.md (hunk 2 above)
Comment (bbadaboom) [reply docs/firmware-flashing.md]: Thanks for the correction! Its hard to indicate where OpenOCD is located, as it depends highly on the installation method and OS. But I tried to improve the wording.

**Pull request merged**